pkg/util/azureclient/fake/virtualmachine.go

```diff
@@ -0,0 +1,81 @@
+package fake
```

Comment:
would like to see a direct map between the name of the client and the filename

Reply:
> don't love the naming - I definitely prefer the idea of kubeclient.Interface, etc.
> sorry for the late review on this.

I don't mind, but the only down side to this is
```
type Upgrader interface {
// other methods here..
	kubeclient.Interface
}

type Upgrade struct {
	kubeclient.Interface
// other stuff here
}

// usage
func (u *Upgrade) foo() {
  // so we can only have one member interface (not sure if that is the correct name) as the 
 // "Interface" name will clash, also u.Interface sounds like it is an interface on "u", not kubeclient..
  u.Interface.kubeFunctionX()
}
```